### PR TITLE
Add label support to the cronjobs.

### PR DESCRIPTION
**Pull request by singularo**
Needed for other MR in https://github.com/universityofadelaide/shepherd/

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -625,8 +625,8 @@ public function deleteRoute(string $name) {
   /**
    * {@inheritdoc}
    */
-  public function getBuilds(string $name) {
-    return $this->apiCall(__METHOD__, '', 'buildconfig=' . $name);
+  public function getBuilds(string $name, string $label) {
+    return $this->apiCall(__METHOD__, '', $name);
```
Comment (bart88) [inline src/Client.php]: Err what ? Use the name as the label ?? Can we have an explanation for what this is doing

File: src/Client.php [hunk 2]
```diff
@@ -1099,8 +1099,8 @@ public function deleteDeploymentConfig(string $name) {
   /**
    * {@inheritdoc}
    */
-  public function getCronJob(string $name) {
-    return $this->apiCall(__METHOD__, $name);
+  public function getCronJob(string $name, string $label = NULL) {
+    return $this->apiCall(__METHOD__, $name, $label);
```
Comment (bart88) [inline src/Client.php]: But what if this is NULL ?

File: src/Client.php [hunk 3]
```diff
@@ -1146,8 +1147,22 @@ public function updateCronJob(string $name, string $image_name, string $schedule
...
    */
-  public function deleteCronJob(string $name) {
-    return $this->apiCall(__METHOD__, $name);
+  public function deleteCronJob(string $name, string $label = NULL) {
+    // If the name was passed in, just delete that specific one.
+    if (!empty($name)) {
+      return $this->apiCall(__METHOD__, $name);
+    }
+
+    // If there was no name, but is a label, delete all jobs that match.
```
Comment (bart88) [inline src/Client.php]: So this also deletes all jobs not just a singular one...

File: src/Client.php (hunk 1 above)
Comment (pingers) [reply src/Client.php]: Guessing it's supposed to be $label

**Comment by pingers** [comment]
I'm really leaning towards passing in a key => value array and just describing the possible array structure. This is done throughout Drupal core for a number of things and seems to work out better than multiple optional parameters. Any thoughts? I'm not set on making the change.

**Comment by singularo** [comment]
Meh, I don't mind if that changes, I just want it deployable and working. MVP!

File: src/Client.php (hunk 3 above)
Comment (singularo) [reply src/Client.php]: Yep, when its a label, all jobs that match, gone. Pfft.

**Comment by singularo** [comment]
This seems to be right now.

**Comment by pingers** [comment]
:shipit:

**Pull request merged**